### DISCUSSION TITLE
Transitions from a playlist to an input.harbor

### DISCUSSION THREAD

**Discussion by SkillenUK** [savonet/liquidsoap#2990] [General]
Hi all. I know I've posted asking for help a few times over the last week so apologies for that, this should be the last one for a while! I've searched through various posts (discussions and issues) and tried differnet things, but since moving to liquidsoap version 2.x I just can't get my playlist to fade out when switching to the live input.harbor. I didn't post this as an issue because I'm not sure if it should be possible, I suspect that it's down to the way which I am trying to do it. My script is something like this...
```

autodj = playlist(mode="randomize",reload=1,reload_mode="rounds","/path/to/autodj.m3u")

djlive = input.harbor("live",port=9010,password="hackme",buffer=2.)
djlivessl = input.harbor.ssl("live",port=9020,auth=dj_auth,buffer=2.)

live = blank.strip(max_blank=10.,mksafe(fallback(track_sensitive=false, [djlivessl,djlive])))

def tolive(old,new)
  sequence(merge=true,[fade.out(duration=1.,(old:source)),blank(duration=1.),(new:source)])
end

def toauto(old,new)
  sequence(merge=true,[blank(duration=2.),fade.in(duration=1.,(new:source))])
end

output =  mksafe(fallback(track_sensitive=false, transition_length=2., transitions=[tolive,toauto], [(live:source),(autodj:source)]))
```

Another variation which I have tried is...

```
def tolive(old,new)
  add(normalize=false, [fade.out(duration=1.,old),sequence([blank(duration=1.), (new:source)]),])
end
```

The transition from the input.harbor back to the playlist (toauto) is working because I added the blank in to the sequence. I understand the issues with transitions and the input.harbor buffer, which will be worse in my case as I set it back to 2. I didn't like the extra latency that the bigger default buffer of 12 was adding. But I think that the short fade out on the playlist before the input.harbor starts should work (tolive). I didn't expect to need the blank but I added it while trying different things.

I was always able to do this in liquidsoap 1.x with the following definition...

```
def tolive(old,new) =
  sequence([fade.final(duration=1.,old),new])
end
```

Hopefully someone can point me in the right direction or let me know that it's not possible so that I don't continue tinkering with it. I'm just trying to avoid the abrupt end to the playlist when a DJ goes live by adding a quick fade out. Thanks,

Skillen.

**Comment by toots** [comment]
Switch-based transitions are marked for deprecation. I'm working on a documentation write-up on how to easily migrate away from it and will report here.

**Reply by desavil** [reply]
When can I expect documentation?
